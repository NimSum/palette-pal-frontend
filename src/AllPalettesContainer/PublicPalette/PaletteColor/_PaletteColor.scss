.PaletteColor {
	width: 100%;
	min-width: 150px;
	height: 40px;
	display: flex;
	align-items: center;
	justify-content: center;

	.PhotoshopPicker {
		width: 128% !important;
		z-index: 500;
		margin-bottom: 30px;
		position: absolute;
		top: 15px;
		z-index: 500;

		span,
		input,
		div {
			font-size: 11px !important;
		}

		div.flexbox-fix {
			height: 262px;

			div:first-child,
			div:nth-child(2) {
				max-height: 230px;
			}
			div {
				max-height: 250px;

				div {
					max-height: 200px;
				}
			}
		}
	}

	.color-hex {
		font-size: 14px;
		opacity: .6;
		width: 100%;

		.fa-pencil-alt {
			opacity: 0;
			position: absolute;
			right: 18%;
			margin-top: 3px;
			cursor: pointer;
		}

		&:hover {
			.fa-pencil-alt {
				opacity: 1;
			}
		}
	}
}
