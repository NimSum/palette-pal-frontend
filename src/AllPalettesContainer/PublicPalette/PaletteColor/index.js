import React, { Component } from 'react';
import colorFormatter from '../../../utils/colorFormatter';
import colorContraster from '../../../utils/colorContraster';

class PaletteColor extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    return (
      <div
        className="PaletteColor"
        key={this.props.id}
        style={{ backgroundColor: this.props.color }}
      >
        <p
          className="color-hex"
          style={{ color: colorContraster(this.props.color) }}
        >
          {colorFormatter(this.props.color, this.props.format)}
        </p>
      </div>
    );
  }
}

export default PaletteColor;