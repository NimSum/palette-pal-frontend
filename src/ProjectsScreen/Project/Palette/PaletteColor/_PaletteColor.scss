 .PaletteColor {
    width: 100%;
    min-width: 150px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;

    .PhotoshopPicker {
      position: absolute;
      top: 15px;
      z-index: 500;
    }

    .color-hex {
      font-size: 14px;
      opacity: .6;
      width: 100%;

      .fa-pencil-alt {
        opacity: 0;
        position: absolute;
        right: 18%;
        margin-top: 3px;
        cursor: pointer;
      }

      &:hover {
        .fa-pencil-alt {
          opacity: 1;
        }
      }
    }
  }