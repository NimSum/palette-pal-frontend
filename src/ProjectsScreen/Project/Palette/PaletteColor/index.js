import React, { Component } from 'react';
import { PhotoshopPicker } from 'react-color';
import colorFormatter from '../../../../utils/colorFormatter';
import colorContraster from '../../../../utils/colorContraster';

class PaletteColor extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showPicker: false,
      update: ''
    }
  }

  handleChangeComplete = color => {
    this.setState({ update: color.hex })
  }

  updateColor = () => {
    const { palette_id, palette_name, project_id } = this.props;
    this.props.updatePaletteData({id: palette_id, [this.props.id]: this.state.update, project_id, palette_name}, 'update');
    this.setState({ update: '', showPicker: false })
  }

  render() {
    const picker = this.state.showPicker ? (
      <PhotoshopPicker
        header="Choose new color"
        className="PhotoshopPicker"
        color={this.state.update || this.props.color}
        onChangeComplete={this.handleChangeComplete}
        onAccept={this.updateColor}
        onCancel={() => this.setState({ showPicker: !this.state.showPicker, update: '' })}
      />
    ) : null;

    return (
      <div
        className="PaletteColor"
        key={this.props.id}
        style={{ backgroundColor: this.props.color }}
      >
        {picker}
        <p
          className="color-hex"
          style={{ color: colorContraster(this.props.color) }}
        >
          {colorFormatter(this.props.color, this.props.format)}
          <i
            className="fas fa-pencil-alt"
            onClick={() => this.setState({showPicker: !this.state.showPicker})}
          />
        </p>
      </div>
    );
  }
}

export default PaletteColor;