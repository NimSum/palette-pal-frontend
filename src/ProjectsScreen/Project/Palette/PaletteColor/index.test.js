import React from 'react';
import PaletteColor from './index';
import { shallow } from 'enzyme';

describe('PaletteColor', () => {
  let wrapper;
  let instance;
  const mockUpdatePaletteData = jest.fn();

  beforeEach(() => {
    wrapper = shallow(< PaletteColor
      color={'#454545'}
      palette_id={3}
      palette_name={'test palette'}
      id={`color_1`}
      project_id={2}
      updatePaletteData={mockUpdatePaletteData}
      format={'hex'}
    />)

    instance = wrapper.instance();

  })
  
  it('shoud match component PaletteColor snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the expected default state', () => {
    expect(wrapper.state()).toEqual({
      showPicker: false,
      update: ''
    })
  })

  it('should set the state property showPicker to true when the edit icon is clicked', () => {
    expect(wrapper.state('showPicker')).toEqual(false);

    wrapper.find('.fa-pencil-alt').simulate('click');

    expect(wrapper.state('showPicker')).toEqual(true);
  })

  it('should update the update property of state when handleChangeComplete is called', () => {
    expect(wrapper.state('update')).toEqual('');

    instance.handleChangeComplete({hex: '#343434'})

    expect(wrapper.state('update')).toEqual('#343434');
  })

  it('should call updatePaletteData and reset state to default when updateColor is called', () => {
    expect(wrapper.state('update')).toEqual('');
    instance.handleChangeComplete({ hex: '#343434' })
    expect(wrapper.state('update')).toEqual('#343434');

    instance.updateColor();

    expect(wrapper.state('update')).toEqual('');
    expect(mockUpdatePaletteData).toHaveBeenCalled();
  })
})