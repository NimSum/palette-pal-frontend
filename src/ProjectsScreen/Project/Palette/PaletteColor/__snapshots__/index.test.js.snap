// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteColor shoud match component PaletteColor snapshot 1`] = `ShallowWrapper {}`;
